pkg/webhooks/resource/validation/validation.go

```diff
@@ -120,20 +120,19 @@ func (v *validationHandler) HandleValidation(
 		return false, webhookutils.GetBlockedMessages(engineResponses), nil
 	}
 
-	go v.handleAudit(ctx, policyContext.NewResource(), request, policyContext.NamespaceLabels(), engineResponses...)
```

Comment:
We still want to process audit policies asynchronously, ideally we need to add audit warn policies to the policy cache and retrieve and apply them during admission review:

https://github.com/kyverno/kyverno/blob/fc316aca858d3f7cf5ef715254100bbce2f80be4/pkg/webhooks/resource/handlers.go#L197

Reply:
I did not update the policy cache to keep the code more simpler so that there is still a clear boundary between audit and enforce

Reply:
The policy cache should be used for all policies to be executed in the admission controller.....not just for enforce mode policies. 

We can add methods like `Policy.AdmissionReview()` which takes which returns true for policies that should be executed during admission review, or `PolicyCache.Get(policyType, admissionReview)` which returns the appropriate policy set from cache.